GoogleSignIn/Sources/GIDSignIn.m

```diff
@@ -997,15 +995,31 @@ - (void)assertValidPresentingViewController {
   }
 }
 
+#if TARGET_OS_IOS && !TARGET_OS_MACCATALYST
 // Asserts the requested scopes are valid.
 - (void)assertValidScopes:(NSArray<NSString *> *)scopes {
-  if ([scopes containsObject:@"https://www.googleapis.com/auth/verified.age.over18.standard"]) {
+  NSDictionary *scopeToClassMapping = @{
```

Comment:
Defining this mapping here will mean it is recreated with each method call. That's not necessary. We can make this cleaner. I see a couple of ways forward.

1) Define this dictionary as a private property or as a constant in this file.

2) Create some kind of "restricted scopes registry" (e.g., `GIDRestrictedScopesRegistry`) that keeps track of restricted scopes (i.e., scopes that should not be passed here and require their own flows). `GIDRestrictedScopesRegistry` would hold a set of scopes whose use are restricted from being passed into `-[GIDSignIn addScopes...]` and therefore require special handling. `GIDRestrictedScopesRegistry` would have properties for the set of restricted scopes and a mapping from restricted scope to class. It would also have methods to i) determine if a supplied scope is restricted and ii) return the restricted scopes from a passed set of scopes. These two methods could also return something (a custom object?) that provides the class that should be used for the scope's flow.

Option 1 is faster. Option 2 is more scalable and testable. My preference is for option 2 since it isn't that much more work.

Reply:
I could go for either though; so, it's up to you!